truffleruby/src/main/ruby/post-boot/post-boot.rb

```diff
@@ -16,6 +16,27 @@
     require 'rubygems'
   rescue LoadError
   else
+    unless Gem.dir.include?(Truffle::Boot.ruby_home)
```

Comment:
I think we should we check the whole `Gem.path`.
Otherwise RubyGems might still load gems from another Ruby.